backend/src/channels/channels.service.ts

```diff
@@ -55,11 +77,9 @@ export class ChannelsService {
     channel.name =
       '-' + createChannelDto.userOneId + '-' + createChannelDto.userTwoId;
     if (
-      await this.channelsRepository.findOne({
-        where: {
-          type: ChannelType.DIRECT,
-          name: channel.name,
-        },
+      await this.channelsRepository.findOneByOrFail({
+        type: ChannelType.DIRECT,
+        name: channel.name,
       })
     )
       throw 'Direct channel already exists.';
```

Comment:
```suggestion
   try {
      await this.channelsRepository.findOneByOrFail({
        type: ChannelType.DIRECT,
        name: channel.name,
      })
      throw 'Direct channel already exists.';
    } catch {}  
```

Reply:
Why ? Don't we catch it in the controller ?

Reply:
If I am correct, you want to continue with the flow of your function if the direct channel does not exist. Your `if` condition will throw in this case, stopping the execution of `createDirectChannel` and `create` with no creation of a channel in the database.

Reply:
You're absolutely correct, my bad; but given that the name is defined as Unique in the channel entity, wouldn't it be better to simply delete this check altogether ? In the new version i actually save in the checkDirectUser, renamed createDirectChannel